server/src/main/scala/com/ossuminc/riddl/lsp/server/RiddlLSPTextDocumentService.scala

```diff
@@ -149,7 +135,7 @@ class RiddlLSPTextDocumentService extends TextDocumentService {
     val selectedItem: Seq[Messages.Message] =
       lineItems
         .map { items =>
-          val filtered = items.filterNot(_.loc.offset == charPosition)
+          val filtered = items.filter(_.loc.col == charPosition)
```

Comment:
charPosition comes from the parameter of the request. lineItems is derived from the AST saved globally, and is not supposed to come any other way.

There was no mistmatch here due to global data, I simply did not check thoroughly the values of offset v col until today, and so assumed wrong that offset was the correct to use. I was in the middle of figuring out more complex things at the time I originally wrote using offset, and forgot to come back to this and ensure it was the right field to use.

Reply:
Okay. Not thrilled with the use of global data, mostly because it is bad style and produces these kinds of discussions. Find a way to eliminate it if you can. 

Your second paragraph is what I originally though to address.  So, that change is fine, BUT it is incomplete unless you also provide a check on _.loc.line as well.  Since "charPosition" is relative to the start of a line, you need to make sure you are also on the right line number.

Reply:
`lineItems` is the result of filtering all after grouping by `_.loc.line`

For global vars, I am not sure what else to do except introduce an Actor system. I brought this up briefly and don't really feel I have the expertise to tell if there is a good alternative for managing state here.

Reply:
Actors are not a solution for this. Way too heavyweight. When I get back let's have a discussion about this functional programming. You can resolve this for now.